### PR TITLE
Install with update repo

### PR DESCRIPTION
**Pull request by nilxam**
https://progress.opensuse.org/issues/9620

### CODE REVIEW

**Comment by nilxam** [comment]
relevant needles are in another PR https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/49

File: tests/installation/setup_online_repos.pm [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        send_key "down";
+    }
+
+    if (get_var("WITH_UPDATE_REPO")) {
+        assert_screen 'update-repos-selected', 10;
+    }
+    # TODO: assert screen for the rest of repos in case they are enabled
+
+    send_key $cmd{'next'};    # Next
+}
```
Comment (coolo) [inline tests/installation/setup_online_repos.pm]: Looks like we need to run the perlcritic check also on the tests to get rid of new code using quotes :)

**Comment by nilxam** [comment]
ah, $cmd{''}, I was copied it from somewhere else and did not aware the quote issue, fixed and rebased.

File: tests/installation/logpackages.pm [hunk 2]
```diff
@@ -5,7 +5,12 @@ use testapi;
 
 sub run() {
     my $self = shift;
-    assert_screen "before-package-selection";
+
+    my $timeout = 20;
+    if (get_var("HAVE_ADDON_REPOS")) {
+        $timeout = 200;    # need spend time writing list of online repos
+    }
+    assert_screen "before-package-selection", $timeout;
```
Comment (coolo) [inline tests/installation/logpackages.pm]: shouldn't this timeout be in the previous test? It's rather suprising here

Comment (nilxam) [inline tests/installation/logpackages.pm]: after clicked next, the installer trying to write the final repos setup, it causes some time to finish it, and once it finished, it is changed to desktop-selection directly, no summary or anything else we can ensure the repos setup is finished...

or I can removing this part and check desktop-selection again at the end of setup_online_repos.pm?

**Comment by nilxam** [comment]
like this

**Comment by coolo** [comment]
yeah, thanks. that's much clearer

**Pull request merged**